native/core/src/data_type.rs

```diff
@@ -1,241 +0,0 @@
-}
-
-impl StringView {
-    pub fn as_utf8_str(&self) -> &str {
-        unsafe {
-            let slice = std::slice::from_raw_parts(self.ptr as *const u8, self.len as usize);
-            std::str::from_utf8_unchecked(slice)
-        }
-    }
-}
-
-impl Default for StringView {
-    fn default() -> Self {
-        Self {
-            len: 0,
-            prefix: [0; STRING_VIEW_PREFIX_LEN],
-            ptr: 0,
-        }
-    }
-}
-
-impl PartialEq for StringView {
-    fn eq(&self, other: &Self) -> bool {
-        if self.len != other.len {
-            return false;
-        }
-        if self.prefix != other.prefix {
-            return false;
-        }
-        self.as_utf8_str() == other.as_utf8_str()
-    }
-}
-
-pub trait NativeEqual {
-    fn is_equal(&self, other: &Self) -> bool;
-}
-
-macro_rules! make_native_equal {
-    ($native_ty:ty) => {
-        impl NativeEqual for $native_ty {
-            fn is_equal(&self, other: &Self) -> bool {
-                self == other
-            }
-        }
-    };
-}
-
-make_native_equal!(bool);
-make_native_equal!(i8);
-make_native_equal!(i16);
-make_native_equal!(i32);
-make_native_equal!(i64);
-make_native_equal!(i128);
-make_native_equal!(StringView);
-
-impl NativeEqual for f32 {
-    fn is_equal(&self, other: &Self) -> bool {
-        self.total_cmp(other) == cmp::Ordering::Equal
-    }
-}
-
-impl NativeEqual for f64 {
-    fn is_equal(&self, other: &Self) -> bool {
-        self.total_cmp(other) == cmp::Ordering::Equal
-    }
-}
-pub trait NativeType: Debug + Default + Copy + NativeEqual {}
-
-impl NativeType for bool {}
-impl NativeType for i8 {}
-impl NativeType for i16 {}
-impl NativeType for i32 {}
-impl NativeType for i64 {}
-impl NativeType for i128 {}
-impl NativeType for f32 {}
-impl NativeType for f64 {}
-impl NativeType for StringView {}
-
-/// A trait for Comet data type. This should only be used as generic parameter during method
-/// invocations.
-pub trait TypeTrait: 'static {
-    type Native: NativeType;
-    fn type_kind() -> TypeKind;
-}
-
-macro_rules! make_type_trait {
-    ($name:ident, $native_ty:ty, $kind:path) => {
-        pub struct $name {}
-        impl TypeTrait for $name {
-            type Native = $native_ty;
-            fn type_kind() -> TypeKind {
-                $kind
-            }
-        }
-    };
-}
-
-make_type_trait!(BoolType, bool, TypeKind::Boolean);
-make_type_trait!(ByteType, i8, TypeKind::Byte);
-make_type_trait!(ShortType, i16, TypeKind::Short);
-make_type_trait!(IntegerType, i32, TypeKind::Integer);
-make_type_trait!(LongType, i64, TypeKind::Long);
-make_type_trait!(FloatType, f32, TypeKind::Float);
-make_type_trait!(DoubleType, f64, TypeKind::Double);
-make_type_trait!(DecimalType, i128, TypeKind::Decimal);
-make_type_trait!(StringType, StringView, TypeKind::String);
-make_type_trait!(BinaryType, StringView, TypeKind::Binary);
-make_type_trait!(TimestampType, i64, TypeKind::Timestamp);
-make_type_trait!(DateType, i32, TypeKind::Date);
```

Comment:
I forgot that why @sunchao added these traits.

Reply:
oops this was for an experiment to introduce Comet specific array vectors, in order to make vectorized hash table in aggregation work ..

Reply:
Okay, so I guess that we can remove them now?

Reply:
yea, they are not used anymore

Reply:
Thanks for confirming it.